sympy/physics/quantum/qapply.py

```diff
@@ -1,206 +1,364 @@
+
+from sympy.physics.quantum.lapply import (lapply,                 #-- correct path once location of lapply is final
+    # lapply is object agnostic. All object specific information has been
+    # isolated in five Dispatcher handler families:
+    lapply1_type, lapply1_pow_type, c_nc_ncef,
+    lapply_mul2types_A, lapply_mul2types_B,
+    # import lapply support functions that may be used in the handlers here
+    lapply_mul2, lapply_mul2elems, lapply_no_act)
+
+
+__all__ = [
+    'qapply'
+]
+
+#--------+---------+---------+---------+---------+---------+---------+---------
+# qapply function header and doc string
+#
+# Options of lapply are listed for completeness. Options taken from expand()
+# have same names (mul, power_xxx, tensorproduct)
+#------------------------------------------------------------------------------
+
+def qapply(e:Expr, ip_doit=True, dagger=False, op_join=True, tensorproduct=None,
+                   mul=True, power_base=None, power_exp=False, nested_exp=None,
+                   apply_exp=False, cache_for_pow=None, **options ) -> Expr:
+    """Knows about quantum objects and invokes their methods _apply_operator
+    and _apply_from_right_to. Applies SymPy standards to apply resp. multiply
+    all other objects in an expression.
+
+    Parameters
+    ==========
+
+    e : Expr
+        The expression containing operators, states resp. factors, summands and
+        powers. The expression tree will be walked and operators resp.
+        factors and powers applied and distributed over summands.
+    options : dict
+        A dict of key/value pairs that determine how the operator actions are
+        carried out. All options are passed to operator methods
+        like _apply_operator_<rhs>() or _apply_from_right_to_<lhs>().
+
+        The following options modify the action of qapply itself:
+
+        * ``dagger``: if a * b doesn't compute, also try ``Dagger(Dagger(b)*Dagger(a))``
+          (default: False).
+        * ``ip_doit``: call ``.doit()`` in inner products when they are
+          encountered (default: True).
+        * ``op_join``: rewrite ket * bra as outer product ``OuterProduct(ket, bra)``
+          and avoid breaking them up to ket * bra (default: True).
+        * ``mul``: distribute commutative factors over sums. Corresponds to option
+          ``mul`` of ``expand``. ``mul=False`` will do the minimal application only.
+            Use when expansion is slow (default: True).
+        * ``tensorproduct``: Expand sums in factors of ``TensorProduct`` into sums of
+          ``TensorProduct`` (same option as in ``expand``) (default: value of option ``mul``).
+        * ``power_base``: for commutative factors, split powers of multiplied bases
+          (same option as in ``expand``) (default: value of option ``mul``).
+        * ``power_exp``: for cummutative factors, expand addition in exponents into
+          multiplied bases (same option as in ``expand``) (default: False).

```

Comment:
The indentation here is not consistent in all items of the list.

Reply:
Great. Fixed it!

Still ``lapply`` documentation is nowhere to be found in the documentation, also not listed in the physics/quantum functions where I have put it until its final location in the code tree has been determined.
(P.S.: Every time when I visit the documentation preview link, the system starts a full test run for nothing)